ndcube/ndcube.py

```diff
@@ -1330,53 +1331,62 @@ def squeeze(self, axis=None):
-        The value to replace masked data with.
-    unmask: `bool`, optional
-        If True, the newly filled masked values are unmasked. If False, they remain masked
-        Default=False
-    uncertainty_fill_value: `numbers.Number` or scalar `astropy.unit.Quantity`, optional
-        The value to replace masked uncertainties with.
-    fill_in_place: `bool`, optional
-        If `True`, the masked values are filled in place.  If `False`, a new instance is returned
-        with masked values filled.  Default=False.
-    """
-    if fill_in_place:
-        new_data = self.data
-        new_uncertainty = self.uncertainty
-        # Unmasking in-place should be handled later.
-    else:
-        new_data = copy.deepcopy(self.data)
-        new_uncertainty = copy.deepcopy(self.uncertainty)
-        new_mask = False if unmask else copy.deepcopy(self.mask)
-
-
-    masked = False if (self.mask is None or self.mask is False or not self.mask.any()) else True
-    if masked:
-            idx_mask = slice(None) is self.mask is True else self.mask # Ensure indexing mask can index the data array.
-            kwargs["data"][self.mask] = fill_value # Boolean indexing in Python.
-        # else, do nothing to the data.
-
-        # if unmask is True, do: change the True mask values to False, otherwise, do nothing to the mask.
+    def fill_masked(self, fill_value, unmask=False, uncertainty_fill_value=None, fill_in_place=False):
+        """
+        Replaces masked data values with input value.
 
-        if uncertainty_fill_value is not None:
-            if not self.uncertainty:
-                raise TypeError("Cannot fill uncertainty as uncertainty is None.")
-            new_uncertainty.array[idx_mask] = uncertainty_fill_value
+        Returns a new instance or alters values in place.
 
-    if not fill_in_place:
-        # Create kwargs dictionary and return a new instance.
-    elif unmask:
-        self.mask = False
+        Parameters
+        ----------
+        fill_value: `numbers.Number` or scalar `astropy.units.Quantity`
+            The value to replace masked data with.
+        unmask: `bool`, optional
+            If True, the newly filled masked values are unmasked. If False, they remain masked
+            Default=False
+        uncertainty_fill_value: `numbers.Number` or scalar `astropy.units.Quantity`, optional
+            The value to replace masked uncertainties with.
+        fill_in_place: `bool`, optional
+            If `True`, the masked values are filled in place.  If `False`, a new instance is returned
+            with masked values filled.  Default=False.
+        """
+        # variable creations for later use.
+        # If fill_in_place is true, do: assign data and uncertainty to variables.
+        if fill_in_place:
+            new_data = self.data
+            new_uncertainty = self.uncertainty
+            new_mask = False if unmask else self.mask
+            new_unit = self.unit
```

Comment:
The unit should stay the so doesn't need to be set here or in the `else` statement below.  Instead, I meant that, after this `if...else` statement, you should check whether `fill_value` has a unit and then convert it to the same unit as the cube, as then strip the unit, i.e.
```pythion
if hasattr(fill_value, "unit")`:
    fill_value = fill_value.to_value(self.unit)
```

You might need to do something equivalent for `uncertainty_fill_value`?